### PR TITLE
Set up Redux Store for Advisors

### PR DESCRIPTION
**Pull request by andrewjleung**
https://trello.com/c/sxNN1KLs

## Why

We need to draft out the design of the Redux store for the advisor work flow as we start getting into implementing it.

## What

- Create an `AdvisorState` interface containing basic session data along a list of students for use in both the students tab and when searching students to assign a template to
- Add the `advisorReducer` along with advisor actions

## Notes

We are currently settling with not persisting data like notifications or templates within the Redux state and requesting them as needed since they are only used in a single location each. Students on the other hand will be cached in Redux after the first request so it can be reused across both the students tab and when assigning templates to students.

Please let us know if there are any concerns with this approach!

### CODE REVIEW

File: frontend/src/state/reducers/advisorReducer.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  setName,
+  setStudents,
+  setToken,
+  setUserId,
+} from "../actions/advisorActions";
+
+export interface Student {
+  readonly email: string;
+  readonly name: string;
+}
```
Comment (andrewjleung) [inline frontend/src/state/reducers/advisorReducer.ts:16]: This simply contains information needed to display a list of users to the advisor, where the student's email uniquely defines them. 

The idea is that when an advisor selects a student, this email is used to fire off a new request to retrieve the rest of that student's information.

File: frontend/src/state/actions/advisorActions.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { createAction } from "typesafe-actions";
+import { Student } from "../reducers/advisorReducer";
+
+export const setEmail = createAction("advisor/SET_EMAIL", (email: string) => ({
```
Comment (NilsBacke) [inline frontend/src/state/actions/advisorActions.ts:4]: We might run into some issues if we use this action in a file that also uses the equivalent "user" action. Maybe `setAdvisorEmail` is better, but not sure if we'll run into this issue.

**Comment by NilsBacke** [comment]
This is great. We might want to rename the existing user actions and reducer to student actions and reducer (in a future PR)

**Review by NilsBacke — APPROVED** [review]

**Comment by NilsBacke** [comment]
[Start Setting up Redux Store for Advisors](https://trello.com/c/sxNN1KLs/169-start-setting-up-redux-store-for-advisors)

**Pull request merged**